cmd/lakectl/cmd/runs_show.go

```diff
@@ -0,0 +1,104 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/go-openapi/swag"
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/api/gen/models"
+	"github.com/treeverse/lakefs/cmdutils"
+	"github.com/treeverse/lakefs/uri"
+)
+
+const actionRunResultTemplate = `
+Run ID: {{ .RunID }}
+	Event: {{ .EventType }}
+	Branch: {{ .Branch }}
+	Start time: {{ .StartTime }}
+	End time: {{ .EndTime }}
+{{if .CommitID}}	Commit ID: {{ .CommitID }}{{end}}
+	Status: {{ .Status }}
+`
+
+const actionTaskResultTemplate = `{{ $log := .HookLog }}
+{{ range $val := .Hooks }}
+Hook Run ID: {{ $val.HookRunID }}
+	Hook ID: {{ $val.HookID }}
+	Start time: {{ $val.StartTime }}
+	End time: {{ $val.EndTime }}
+	Action: {{ $val.Action }}
+	Status: {{ $val.Status }}
+Output:
+{{ printf $val.HookRunID | call $log }}
+{{ end }}{{ .Pagination | paginate }}
+`
+
+const runsShowRequiredArgs = 2
+
+var runsShowCmd = &cobra.Command{
+	Use:     "show",
+	Short:   "Show run results",
+	Long:    `Show information about the run and all the hooks that were executed as part of the run`,
+	Example: "lakectl actions runs show lakefs://<repository> <run_id>",
+	Args: cmdutils.ValidationChain(
+		cobra.ExactArgs(runsShowRequiredArgs),
+		cmdutils.FuncValidator(0, uri.ValidateRepoURI),
+	),
+	Run: func(cmd *cobra.Command, args []string) {
+		amount, _ := cmd.Flags().GetInt("amount")
+		after, _ := cmd.Flags().GetString("after")
+		u := uri.Must(uri.Parse(args[0]))
+		runID := args[1]
+
+		client := getClient()
+		ctx := context.Background()
+
+		// run result information
+		runResult, err := client.GetRunResult(ctx, u.Repository, runID)
+		if err != nil {
+			DieErr(err)
+		}
+		Write(actionRunResultTemplate, runResult)
+
+		// iterator over hooks - print information and output
+		response, pagination, err := client.ListRunTaskResults(ctx, u.Repository, runID, after, amount)
+		if err != nil {
+			DieErr(err)
+		}
+		data := struct {
+			Hooks      []*models.HookRun
+			HookLog    func(hookRunID string) (string, error)
+			Pagination *Pagination
+		}{
+			Hooks: response,
+			HookLog: func(hookRunID string) (string, error) {
```

Comment:
I think this has the potential to mess stdout and make the command less accessible. Maybe just log the actual command for fetching the output?
`lakectl cat-hook-output lakefs://u.Repository runID hookRunID `

Reply:
Thought about it - this one after discussion with Oz.
Keeping the `cat-hook-output` hidden command and have a simple one command to get all the information.
so the end-user life will be easy